go/interpreter/lfvm/instructions_test.go

```diff
@@ -666,83 +666,46 @@ func TestTransientStorageOperations(t *testing.T) {
 	}
 }
 
-func TestExpansionCostOverflow(t *testing.T) {
-	memTestValues := []uint64{
-		maxMemoryExpansionSize,
-		maxMemoryExpansionSize + 1,
-		math.MaxUint64,
-	}
+func TestGenericDataCopy_ReturnsErrorOn(t *testing.T) {
+	testBuffer := make([]byte, 1024)
 
 	tests := map[string]struct {
-		op         func(*context) error
-		stackSize  int
-		memIndexes []int
-		setup      func(*tosca.MockRunContext)
+		offset        uint256.Int
+		size          uint256.Int
+		gas           tosca.Gas
+		expectedError error
 	}{
-		"mcopy": {
-			op:         opMcopy,
-			stackSize:  3,
-			memIndexes: []int{0, 1, 2},
-			setup:      func(runContext *tosca.MockRunContext) {},
-		},
-		"calldatacopy": {
-			op:         opCallDataCopy,
-			stackSize:  3,
-			memIndexes: []int{0, 2},
-			setup:      func(runContext *tosca.MockRunContext) {},
-		},
-		"codecopy": {
-			op:         opCodeCopy,
-			stackSize:  3,
-			memIndexes: []int{0, 2},
-			setup:      func(runContext *tosca.MockRunContext) {},
-		},
-		"extcodecopy": {
-			op:         opExtCodeCopy,
-			stackSize:  4,
-			memIndexes: []int{0, 2},
-			setup: func(runContext *tosca.MockRunContext) {
-				runContext.EXPECT().AccessAccount(gomock.Any()).AnyTimes().Return(tosca.WarmAccess)
-				runContext.EXPECT().GetCode(gomock.Any()).AnyTimes().Return([]byte{0x01, 0x02, 0x03, 0x04})
-			},
+		"not enough gas": {
+			offset:        *uint256.NewInt(10),
+			size:          *uint256.NewInt(10),
+			gas:           1,
+			expectedError: errOutOfGas,
+		},
+
+		"expansionFailure": {
+			offset:        *uint256.NewInt(math.MaxUint64),
+			size:          *uint256.NewInt(10),
+			gas:           1 << 32,
+			expectedError: errOverflow,
 		},
 	}
 
 	for name, test := range tests {
-		for _, memIndex := range test.memIndexes {
-			for _, memValue := range memTestValues {
-				t.Run(fmt.Sprintf("%v_i:%v_v:%v", name, memIndex, memValue), func(t *testing.T) {
-					ctrl := gomock.NewController(t)
-					runContext := tosca.NewMockRunContext(ctrl)
-					test.setup(runContext)
+		t.Run(name, func(t *testing.T) {
 
-					ctxt := context{
-						params: tosca.Parameters{
-							BlockParameters: tosca.BlockParameters{
-								Revision: tosca.R13_Cancun,
-							},
-						},
-						stack:   NewStack(),
-						memory:  NewMemory(),
-						context: runContext,
-						gas:     12884901899,
-					}
-					ctxt.stack.stackPointer = test.stackSize
-					ctxt.stack.data[memIndex].Set(uint256.NewInt(memValue))
-					for i := range test.memIndexes {
-						if i != memIndex {
-							ctxt.stack.data[i].Set(uint256.NewInt(1))
-						}
-					}
+			ctxt := getEmptyContext()
+			ctxt.gas = test.gas
```

Comment:
```suggestion
ctxt := context{gas: test.gas}
```
only gas and stack are used. nothing else is needed, so it can be changed for this single line.